src/json-ld.net/Core/DocumentLoader.cs

```diff
@@ -54,9 +56,32 @@ public virtual RemoteDocument LoadDocument(string url)
             {
                 throw;
             }
-            catch (Exception)
+            catch (WebException webException)
+            {
+                try
+                {
+                    resp = (HttpWebResponse)webException.Response;
+                    int baseStatusCode = (int)(Math.Floor((double)resp.StatusCode / 100)) * 100;
+                    if (baseStatusCode == 300)
```

Comment:
Shouldn't we explicitly check for redirects here (i.e. not ignore 300, 304 etc?)

Reply:
Perhaps it's not entirely clear from the code, but `(int)(Math.Floor((double)resp.StatusCode / 100)) * 100` will ensure that all status codes between `300` and `399` will be made equal to `300` so it should cover all redirect status codes.

Reply:
Yes but e.g. 300 can't be treated exactly the same as e.g. 301/302 as far as I am aware?

Reply:
Perhaps we should handle 301,302,307 and 308 and just throw NotImplemented otherwise? We could perhaps also handle 300s which contain a Location header.

Reply:
Responses with `300` status code `SHOULD` contain a `Location` header if the server has a preferred choice of representation for the requested resource. Similarly, `301`, `302`, `303`, `307` and `308` have equivalent wording mandating a `Location` header be present. There are a few exceptions such as `304`, `305` and `306`, but they are almost never in use. I think that due to the code checking for the existence of a `Location` header in the case of a status code within the `3xx Redirection` range, this code – although a bit naïve – should be good enough.